version: '3.6'
services:
    db:
        image: mysql
        restart: always
        container_name: mysql8
        environment:
            MYSQL_ROOT_PASSWORD: R_qNhdi5vo
            TZ: Asia/Shanghai
        command:
          # 解决外部无法访问
            --default-authentication-plugin=caching_sha2_password
            --character-set-server=utf8mb4
            --collation-server=utf8mb4_general_ci
            --explicit_defaults_for_timestamp=true
            --lower_case_table_names=1
            --secure_file_priv=/var/lib/mysql
            --sql-mode="STRICT_TRANS_TABLES,NO_ENGINE_SUBSTITUTION,NO_ZERO_DATE,NO_ZERO_IN_DATE,ERROR_FOR_DIVISION_BY_ZERO"
        ports:
            - 3306:3306
        volumes:
            - /etc/localtime:/etc/localtime:ro
            - ./data:/var/lib/mysql
            - ./logs:/var/log
